<div style="font-size: 30px">
    <font color='Blue'> <b>Seasonal Shifts: Understanding Vegetable Price Changes Across Time</b></font> 

<img src="Vegetable-Prices.jpg" style="width:1000px;height:550px"/>

<a id="cont"></a>

## Table of Contents
* <b>[1. Project Overview](#chapter1)
    * [1.1 Introduction](#section_1_1)
        * [1.1.1 Problem Statement](#sub_section_1_1_1)
    * [1.2 Objectives](#section_1_2)
* <b>[2. Importing Packages](#chapter2)
* <b>[3. Loading Data](#chapter3)
* <b>[4. Data Cleaning](#chapter4)
* <b>[5. Exploratory Data Analysis (EDA)](#chapter5)
* <b>[6. Conclusion](#chapter6)</b>
* <b>[7. References](#chapter7)<b>

## 1. Project Overview <a class="anchor" id="chapter1"></a>

### 1.1 Introduction <a class="anchor" id="section_1_1"></a>

Seasonal price variations in vegetables are primarily driven by supply and demand. During the 
peak growing season, when vegetables are abundant, prices tend to be lower. Conversely, in the 
off-season, when production decreases due to unfavourable weather conditions, prices increase. 
This pattern is a fundamental economic principle, but several factors can influence its dynamics. 
Weather plays a crucial role in agriculture. Unseasonably cold or hot weather can delay planting 
or harvesting, reducing the supply of certain vegetables and leading to higher prices. Vegetables 
that are not in season locally may be imported from other regions or countries. The cost of 
transportation, including fuel prices, can significantly affect the final price of these vegetables. 
Consumer preferences can also influence vegetable prices. For example, the demand for certain 
vegetables may increase around specific holidays, leading to temporary price hikes. 
Climate change is increasingly affecting agricultural patterns worldwide, leading to more 
unpredictable weather conditions and impacting the predictability of seasonal price variations. 
Extreme weather events, such as droughts, floods, and hurricanes, can devastate crops, 
reducing supply and driving up prices. Additionally, rising global temperatures can shift the 
growing seasons for many vegetables, affecting their availability and price.

#### 1.1.1 Problem Statement <a class="anchor" id="sub_section_1_1_1"></a>


Vegetable prices are subject to a wide range of factors, including weather conditions, demand, and 
the cost of labour and transportation. However, one of the most significant factors affecting the 
price of vegetables is the season. Seasonal variations can lead to significant fluctuations in the 
availability and cost of different types of vegetables. Understanding these seasonal patterns in 
vegetable prices can help producers make proactive decisions to optimize crop production, 
resource management, marketing strategies, risk mitigation, and financial planning 


#### 1.1.2 Objectives <a class="anchor" id="section_1_2"></a>

1. Price Trend Analysis: Analyze the historical price trends of different vegetables to identify 
patterns and seasonal variations. 
2. Seasonal Impact Assessment: Determine the impact of seasons on vegetable prices and 
explore any notable seasonal price fluctuations. 
3. Correlation Study: Investigate correlations between vegetable prices to understand how 
changes in one vegetable's price may affect others. 
4. Insights Generation: Extract actionable insights and recommendations for stakeholders in 
the agricultural sector to optimize pricing strategies and improve market outcomes. 

## 2. Importing Packages <a class="anchor" id="chapter2"></a>

This data set was created to list all shows available on Netflix streaming, and analyze the data to find interesting facts. This data was acquired in July 2022 containing data available in the United States.

+ For data manipulation and analysis, `Pandas` and `Numpy`.
+ For data visualization, `Matplotlib` and `Seaborn`.


In [1]:
# Libraries for data loading, manipulation and analysis

import numpy as np
import pandas as pd
import csv
import seaborn as sns
import matplotlib.pyplot as plt

# Displays output inline
%matplotlib inline

# Libraries for Handing Errors
import warnings
warnings.filterwarnings('ignore')

## 3. Loading Data <a class="anchor" id="chapter3"></a>

The data used for this project was located in the `titles.csv` file. To better manipulate and analyse the `titles.csv` file, it was loaded into a Pandas Data Frame using the Pandas function, `.read_csv()` and referred to as `titles_import`. For demonstrating the column index in the dataframe , `index_col=False` was implemented.

In [2]:
# loading dataset
prices_df = pd.read_csv("prices.csv", index_col=False)
prices_df.head(2)

Price Dates  Bhindi (Ladies finger)  Tomato  Onion  Potato  Brinjal  Garlic  \
0  01-01-2023                    35.0      18   22.0      20       30      50   
1  02-01-2023                    35.0      16   22.0      20       30      55   

   Peas  Methi  Green Chilli  Elephant Yam (Suran)  
0    25      8          45.0                    25  
1    25      7          40.0                    25

To set the maximum number of columns to be displayed, the `pd.set_option()` function will be put in place.

In [3]:
pd.set_option("display.max_columns", None)

<div class="alert alert-block alert-danger">
<b>To prevent any major unnecessary changes occurring to the original data</b> , a copy of the dataframe will be made using the prices_df.copy() method and referred to as `prices_copy_df`.
</div>

In [5]:
# The copy of the dataframe
prices_copy_df = prices_df.copy()

First we are looking at the shape of the dataframe by using `.shape`.

In [6]:
# Displays the number of rows and columns
prices_copy_df.shape

(287, 11)

**Results** : The dataset consists of 287 rows (observations) and 11 columns (features). 

One of the univariate analysis (non-graphical), is the method `.info()`. This method outputs important details about the dataset which includes the columns: the data types **(Dtype)** of the columns and the count of non-null values.

In [7]:
## Display summary information about the DataFrame.
prices_copy_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 287 entries, 0 to 286
Data columns (total 11 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Price Dates             287 non-null    object 
 1   Bhindi (Ladies finger)  287 non-null    float64
 2   Tomato                  287 non-null    int64  
 3   Onion                   287 non-null    float64
 4   Potato                  287 non-null    int64  
 5   Brinjal                 287 non-null    int64  
 6   Garlic                  287 non-null    int64  
 7   Peas                    287 non-null    int64  
 8   Methi                   287 non-null    int64  
 9   Green Chilli            287 non-null    float64
 10  Elephant Yam (Suran)    287 non-null    int64  
dtypes: float64(3), int64(7), object(1)
memory usage: 24.8+ KB


## 4. Data Cleaning <a class="anchor" id="chapter4"></a>

After using the `info()`, we notice some issues with the format of the Price Dates and prices of vegetables

In [8]:
# Convert 'Price Dates' column from string format to datetime format
# Specify the format of the input date strings as day-month-year
prices_copy_df['Price Dates'] = pd.to_datetime(prices_copy_df['Price Dates'], format='%d-%m-%Y')

**Result:** The column of date strings ('Price Dates') in a DataFrame (prices_copy_df) is converted into datetime objects using a specified date format ('%d-%m-%Y') which results in the column becoming standardized

Standardizing the Price Dates column improves the quality, reliability and usability of out data, which makes it easier to derive meaningful insights and make informed decisions based on time-related information within the dataset

In [9]:
prices_copy_df.head()

Price Dates  Bhindi (Ladies finger)  Tomato  Onion  Potato  Brinjal  Garlic  \
0  2023-01-01                    35.0      18   22.0      20       30      50   
1  2023-01-02                    35.0      16   22.0      20       30      55   
2  2023-01-03                    35.0      16   21.0      20       30      55   
3  2023-01-04                    30.0      16   21.0      22       25      55   
4  2023-01-08                    35.0      16   20.0      21       25      55   

   Peas  Methi  Green Chilli  Elephant Yam (Suran)  
0    25      8          45.0                    25  
1    25      7          40.0                    25  
2    25      7          40.0                    25  
3    25      7          40.0                    25  
4    22      6          35.0                    25

In [10]:
# Specify columns to convert to float
columns_to_convert = ['Tomato', 'Potato', 'Brinjal', 'Garlic', 'Peas', 'Methi',	'Elephant Yam (Suran)']

# Convert selected columns to float
prices_copy_df[columns_to_convert] = prices_copy_df[columns_to_convert].astype(float)

**Result:** The columns Tomato, Potato, Brinjal, Garlic, Peas, Methi, and Elephant Yam (Suran) in the prices_copy_df DataFrame will be converted to float data type


Representing vegetable prices to be in float format offers precision, which is crucial for accurately reflecting decimal values in prices (e.g 1.99 Rand). Using float format ensures that calculations involving prices, such as total costs or average prices, are accurate and free from rounding errors

In [11]:
prices_copy_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 287 entries, 0 to 286
Data columns (total 11 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Price Dates             287 non-null    datetime64[ns]
 1   Bhindi (Ladies finger)  287 non-null    float64       
 2   Tomato                  287 non-null    float64       
 3   Onion                   287 non-null    float64       
 4   Potato                  287 non-null    float64       
 5   Brinjal                 287 non-null    float64       
 6   Garlic                  287 non-null    float64       
 7   Peas                    287 non-null    float64       
 8   Methi                   287 non-null    float64       
 9   Green Chilli            287 non-null    float64       
 10  Elephant Yam (Suran)    287 non-null    float64       
dtypes: datetime64[ns](1), float64(10)
memory usage: 24.8 KB


In [12]:
prices_copy_df.head()

Price Dates  Bhindi (Ladies finger)  Tomato  Onion  Potato  Brinjal  Garlic  \
0  2023-01-01                    35.0    18.0   22.0    20.0     30.0    50.0   
1  2023-01-02                    35.0    16.0   22.0    20.0     30.0    55.0   
2  2023-01-03                    35.0    16.0   21.0    20.0     30.0    55.0   
3  2023-01-04                    30.0    16.0   21.0    22.0     25.0    55.0   
4  2023-01-08                    35.0    16.0   20.0    21.0     25.0    55.0   

   Peas  Methi  Green Chilli  Elephant Yam (Suran)  
0  25.0    8.0          45.0                  25.0  
1  25.0    7.0          40.0                  25.0  
2  25.0    7.0          40.0                  25.0  
3  25.0    7.0          40.0                  25.0  
4  22.0    6.0          35.0                  25.0

The `.describe()` method is a fundamental tool used for generating descriptive statistics